### DISCUSSION TITLE
Improve the Kvrocks's documentation

### DISCUSSION THREAD

**Discussion by git-hulk** [apache/kvrocks#2001] [General]
Many documents in Kvrocks are missing, and we are eager to improve this. But before this, we would like to hear from you, can help to tell us about what you want to know but is absent for now. We will try our best to write it down if possible.

Thank you!

**Comment by jihuayu** [comment]
Many new developers don't know how to use `MakeCmdAttr` to create a new command. such as https://github.com/apache/kvrocks/pull/2018
Function MakeCmdAttr params' means aren't obvious. 

Can we create a blog to talk about it?
After that, when new developers come, we can show them this blog.

**Reply by jihuayu** [reply]
If someone can give me each flag description means, I can write this blog.

**Reply by git-hulk** [reply]
@jihuayu Thank you! I will give you some input if you needed.

**Reply by git-hulk** [reply]
- read-only: this command is read-only
- ok-loading: this command is allowed to run while loading the data, like the info command
- no-multi: this command can't be used inside the multi-exec transaction
- no-script: this command can't be running inside Lua script, like the blocking command: B[L|R]POP
- exclusive: it will use an exclusive lock between workers while running this command
- write: it's a write operation command and cannot run if exceeds the max size of DB

**Comment by jiangyunpeng** [comment]
The [cluster scaling](https://kvrocks.apache.org/docs/cluster/#cluster-scaling) select is too simple.
There are some need improvement point that i thought:
- NOT indicate how to use CLUSTERX MIGRATE.
- NOT indicate how to add new node, Actually first of all need use CLUSTERX SETNODES command to add nodes.
- NOT indicate what difference between  CLUSTERX MIGRATE and CLUSTERX SETSLOT， Actually the primary command is used for migrate slot and the last command is used for update slots info.
- NOT indicate how to execute sync  MIGRATE,  Actually CLUSTERX MIGRATE command support sync MIGRATE.

**Comment by ferbs** [comment]
Thanks for asking! Add a page for kvrocks.conf? 

I found kvrocks.conf in the git repo after noticing it's referenced in the Dockerfile but the file isn't mentioned on the documentation site, not once. 

The file itself is nicely commented but the doc site could perhaps:
* explain its existence and default location
* draw attention to settings that are especially important or frequently modified by users
* explain alternative ways to set config values rather than modifying the .conf file; I discovered some (all?) configs can be passed as command line switches; (and environment variables? I haven't found them yet, would be good to list what's available.)

Even better, the page could provide guidance or link to examples for common scenarios. General ones ranging from high-end to low end would give people a great start:
* heavy: high traffic, high-availability, clustered, relatively big boxes
* medium: modest traffic but HA required; modest hardware but large SSDs; eg company-internal tool or a new SaaS product 
* low: memory & cpu constrained systems; eg CI/CD testing

**Reply by git-hulk** [reply]
Hi @ferbs, thanks for your great suggestions. I will break down those suggestions and add them ASAP.
